x/datapool/keeper/pool_distribute.go

```diff
@@ -0,0 +1,226 @@
+	key := types.GetKeyPrefixSalesHistories(poolID, round)
+	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, key)
+	var histories []*types.SalesHistory
+	for ; iter.Valid(); iter.Next() {
+		history := &types.SalesHistory{}
+		bz := iter.Value()
+		k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, history)
+		histories = append(histories, history)
+	}
+	return histories
+}
+
+// SetInstantRevenueDistribute stores the poolID to which the revenue should be distributed immediately.
+func (k Keeper) SetInstantRevenueDistribute(ctx sdk.Context, instantRevenueDistribute *types.InstantRevenueDistribute) {
+	bz := k.cdc.MustMarshalBinaryLengthPrefixed(instantRevenueDistribute)
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.KeyPrefixInstantRevenueDistribute, bz)
+}
+
+func (k Keeper) GetInstantRevenueDistribute(ctx sdk.Context) *types.InstantRevenueDistribute {
+	store := ctx.KVStore(k.storeKey)
+	if !store.Has(types.KeyPrefixInstantRevenueDistribute) {
+		return &types.InstantRevenueDistribute{}
+	}
+	bz := store.Get(types.KeyPrefixInstantRevenueDistribute)
+	var delayedRevenueDistribute types.InstantRevenueDistribute
+	k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, &delayedRevenueDistribute)
+	return &delayedRevenueDistribute
+}
+
+// addInstantRevenueDistribute adds the poolID to distribute. If there are duplicate poolIDs, they are not added.
+func (k Keeper) addInstantRevenueDistribute(ctx sdk.Context, poolID uint64) {
+	instantRevenueDistribute := k.GetInstantRevenueDistribute(ctx)
+	instantRevenueDistribute.AppendPoolID(poolID)
+	k.SetInstantRevenueDistribute(ctx, instantRevenueDistribute)
+}
+
+func (k Keeper) DistributeRevenuePools(ctx sdk.Context) error {
+	instantRevenueDistribute := k.GetInstantRevenueDistribute(ctx)
+
+	if instantRevenueDistribute.IsEmpty() {
+		return nil
+	}
+
+	// This lastIndex is to execute the following iteration in the next block if it does not end in one block.
+	// But We're currently processing it all in one block. :)
+	lastIndex := 0
+	// TODO We need to think about how to deal with the pool where distribution has failed.
+	for i, poolID := range instantRevenueDistribute.PoolIds {
+		lastIndex = i
+		// search current pool info
+		pool, err := k.GetPool(ctx, poolID)
+		if err != nil {
+			return sdkerrors.Wrap(types.ErrRevenueDistribute, err.Error())
+		}
+
+		// send deposit to curator
+		err = k.sendDepositToCurator(ctx, pool)
+		if err != nil {
+			return err
+		}
+
+		// distribute of revenue to each seller
+		err = k.executeRevenueDistribute(ctx, pool)
+		if err != nil {
+			return err
+		}
+	}
+	// delete this pool from instantRevenueDistribute
+	instantRevenueDistribute.RemovePreviousIndex(lastIndex)
+	k.SetInstantRevenueDistribute(ctx, instantRevenueDistribute)
+
+	return nil
+}
+
+// sendDepositToCurator returns to the Curator if the pool status is ACTIVE but the deposit has not yet been returned.
```

Comment:
```suggestion
// sendDepositToCurator returns the deposit to the Curator if the pool status is ACTIVE but the deposit has not yet been returned.
```

Reply:
OK. I fixed it.
https://github.com/medibloc/panacea-core/pull/318/commits/5438b937521dcc52e21db57680c0c0b560d7dcce